### ISSUE TITLE
Autocomplete Lookups does not work with multi-table inheritance 

### ISSUE THREAD

**Issue by cgabard** [sehmaschine/django-grappelli#873]
We use  django 2.1.3 and django-grappelli 2.12.1 on python 3.7

------

We have an exception on `json/encoder.py`

``` 
TypeError: Object of type User is not JSON serializable
```

coming from the line : https://github.com/sehmaschine/django-grappelli/blob/f5153436ea822c5ef920f27c59083b69881b7084/grappelli/views/related.py#L47

our code is equivalent to this:

```python

class User(models.Model):
    pass

class Foo(User):
    name = models.CharField("name", max_length=200)

    @staticmethod
    def autocomplete_search_fields():
        return ("name__icontains",)

class Bar(models.Model):
    foo = models.ForeignKey(Foo, on_delete=models.SET_NULL)

class BarAdmin(admin.ModelAdmin):
    fields = ("foo", )
    raw_id_fields = ('foo',)

    autocomplete_lookup_fields = {
        'fk': ['foo'],
    }
```

------

After some search, we have isolated that error come from multi-table inheritance.

On the ajax call, that generate the error, JS send the get parameter `to_field=user_ptr` which is the name of the FK created by django to manage the multi-table inheritance. Please note it is the name of the FK field, not the id (which is `user_ptr_id`). So when Bar elements are retrieve, the data value extracted is then the parent User instance (and not their ID which is the primary key of the table). And then the json encoder can not dump it.

On the grappelli js code, this file is extracted from the raw_id_fields generated by django : https://github.com/sehmaschine/django-grappelli/blob/f5153436ea822c5ef920f27c59083b69881b7084/grappelli/static/grappelli/js/grappelli.js#L176 and it seems that for multi-table inheritance, it is the FK field that is provided, not the corresponding id.

-----

I have not a lot of idea how to fix this easily but if I can help...

**Comment by sehmaschine** [comment]
does it work with `raw_id_fields` only (without autocompletes)?

**Comment by cgabard** [comment]
Yes, and it is what we use for now

**Comment by sehmaschine** [comment]
Could you please try this chunk of code with related.py (from line 110) and let me know if that works with your setup.

```
    def get_return_value(self, obj, obj_id):
        to_field = self.GET.get('to_field', None)
        if to_field is not None:
            return getattr(obj, to_field)
            return_value = getattr(obj, to_field)
            if not isinstance(return_value, str) and not isinstance(return_value, int):
                return_value = obj.pk
            return return_value
        return obj_id
```

**Comment by cgabard** [comment]
It  work well for us.

**Comment by sehmaschine** [comment]
Thanks. I'll do a new release asap. Unfortunately, it's not the best workaround, but we can only use the `_to_field` we get with `raw_id_fields`.

**Comment by cgabard** [comment]
Thank you for your reactivity

**Comment by cgabard** [comment]
Hello. Any idea when you will release the new version ? Can I do something to help you ?

**Comment by cgabard** [comment]
Ok, thank.

**Comment by sehmaschine** [comment]
Just released 2.12.2.

**Comment by cgabard** [comment]
Thanks
